src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/csip/CSIPResourceIntTest.kt

```diff
@@ -0,0 +1,211 @@
+package uk.gov.justice.digital.hmpps.nomisprisonerapi.csip
+
+import org.assertj.core.api.Assertions
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.helper.builders.NomisData
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.helper.builders.NomisDataBuilder
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.helper.builders.Repository
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.integration.IntegrationTestBase
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.integration.latestBooking
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.CSIPReport
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.Offender
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.Staff
+
+class CSIPResourceIntTest : IntegrationTestBase() {
+  @Autowired
+  private lateinit var repository: Repository
+
+  @Autowired
+  private lateinit var nomisDataBuilder: NomisDataBuilder
+
+  private lateinit var reportingStaff: Staff
+  private lateinit var offender: Offender
+  private lateinit var csip1: CSIPReport
+  private lateinit var csip2: CSIPReport
+  private lateinit var csip3: CSIPReport
+
+  @BeforeEach
+  internal fun createCSIPReports() {
+    nomisDataBuilder.build {
+      setUpStaffAndOffender()
```

Comment:
not sure extracting this to a method makes it more readable - but this is a personal style thing